File: src/test/scala/Optional.scala
import org.scalatest.funsuite.AnyFunSuite
import ts2mls.TSProgram
import ts2mls.types._

class Optional extends AnyFunSuite {
  test("Optional") {
    val program = TSProgram(Seq("src/test/typescript/Optional.ts"))
    assert(TypeCompare(program.>("buildName"), "(string, string | undefined) => string"))
    assert(TypeCompare(program.>("buildName2"), "(string, string | undefined) => string"))
    assert(TypeCompare(program.>("buildName3"), "(string, any[]) => string"))
  }
}

File: src/test/typescript/Optional.ts
function buildName(firstName: string, lastName?: string) {
    return firstName + lastName;
}

function buildName2(firstName: string, lastName = "DIO") {
    return firstName + lastName;
}

function buildName3(firstName: string, ...lastName) {
    console.log(lastName)
    return firstName;
}

interface SquareConfig {
    color?: string;
    width?: number;
}